src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/DefaultRazorFormattingService.cs

```diff
@@ -85,6 +85,36 @@ public override async Task<TextEdit[]> FormatAsync(Uri uri, RazorCodeDocument co
             return edits;
         }
 
+        public override Task<TextEdit[]> ApplyFormattedEditsAsync(Uri uri, RazorCodeDocument codeDocument, RazorLanguageKind kind, TextEdit[] formattedEdits, FormattingOptions options)
```

Comment:
It seems we're defining this function right now, and it never existed before. If that's the case can we make it not return a Task if we're not using any async methods? Or is there some reason we need to future-proof this?

Reply:
Yes. I am in general making formatting APIs async because it might need to call into the client to format the underlying buffers.